### PR TITLE
Change cfg logic for com1DFA

### PR DESCRIPTION
**Pull request by awirb**
created new workflow:

- com1DFAPreprocess
- com1DFAMain
- com1DFAPostprocess

### CODE REVIEW

File: avaframe/com1DFA/com1DFA.py [hunk 1]
```diff
@@ -51,42 +52,79 @@
...
-    return inputSimFilesAll, cfgFilesRels
+    # read input data files (release, DEM, etc.) and update configuration with this
+    inputSimFilesAll, cfgStart = com1DFATools.setRelThIni(avalancheDir, com1DFA, cfgStart)
 
+    # initialise reportDictList and flag indicating whether simulations have been performed
+    reportDictList = []
+    simsPerformed = False
+    #
+    # # copy inputSimFilesAll - as this becomes changed according to current release scenario
+    # inputSimFiles = inputSimFilesAll.copy()
```
Comment (matthiasto) [inline avaframe/com1DFA/com1DFA.py:101]: do we still need this?

File: avaframe/com1DFA/com1DFA.py [hunk 2]
```diff
@@ -51,42 +52,79 @@
...
+    modCfg, variationDict = dP.getParameterVariationInfo(avalancheDir, com1DFA, cfgStart)
+
+    # first fetch info on already existing simulations in Outputs
+    # if need to reproduce exactly the hash - need to be strings with exactely the same number of digits!!
+    simDFOld, simNameOld = cfgUtils.readAllConfigurationInfo(avalancheDir, specDir='')
+
+    # create a configuration object per simulation to run (from configuration) gathered in simDict
+    # only new simulations are included in this simDict
+    # key is simName and corresponds to one simulation
+    simDict = prepareVarSimDict(modCfg, inputSimFilesAll, variationDict, simNameOld=simNameOld)
```
Comment (matthiasto) [inline avaframe/com1DFA/com1DFA.py:117]: why do we have this as a dict and not a DF?
you use simDFOld, why not also simDFNew?

File: avaframe/com1DFA/com1DFA.py [hunk 3]
```diff
@@ -111,147 +151,148 @@ def com1DFAMain(avalancheDir, cfgMain, cfgFile=''):
...
-        # first get info on already existing simulations in Outputs
-        simDFOld, simNameOld = cfgUtils.readAllConfigurationInfo(avalancheDir, specDir='')
+            # ++++++++++PERFORM com1DFA SIMULAITON++++++++++++++++
+            dem, reportDict, cfgFinal, tCPU, inputSimFilesNEW, particlesList, fieldsList, tSave = com1DFA.com1DFACore(cfg,
+                avalancheDir, cuSim, inputSimFiles, outDir, simHash=simHash)
+            simDF.at[simHash, 'nPart'] = str(int(particlesList[0]['nPart']))
 
-        # prepare simulations to run (only the new ones)
-        simDict = prepareVarSimDict(modCfg, inputSimFiles, variationDict, simNameOld=simNameOld)
+            # TODO check if inputSimFiles not changed within sim
```
Comment (matthiasto) [inline avaframe/com1DFA/com1DFA.py:193]: do we still need the todo?

File: avaframe/com1DFA/com1DFA.py (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:117]: it is a dictionary as it has different types of objects for the keys, the configparser object, strings, floats, ... simDFOld was introduced to distinguish which simulations are already present in the outputs, simDF is the one for the actual simulation run - could be renamed to new, but could stay just simDF too I think

File: avaframe/com1DFA/com1DFA.py (hunk 3 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:193]: good question ;)

File: avaframe/com1DFA/com1DFA.py (hunk 1 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:101]: removed is not needed anymore now

File: avaframe/com1DFA/com1DFA.py [hunk 4]
```diff
@@ -51,42 +52,79 @@
...
+        inputSimFiles: dict
+            dictionary with input files info
+        outDir: str
+            path to store outputs
     """
 
-    # check if thickness settings in ini file are valid
-    for thType in ['entTh', 'relTh', 'secondaryRelTh']:
-        _ = dP.checkThicknessSettings(cfgInitial, thType)
+    modName = 'com1DFA'
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:81]: changed this

File: avaframe/com1DFA/deriveParameterSet.py [hunk 5]
```diff
@@ -724,3 +737,27 @@ def writeToCfgLine(values):
         valString = valString + '|%.12f' % val
 
     return valString
+
+
+def convertToCfgList(parameterList):
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/deriveParameterSet.py:742]: moved to cfgUtils

File: avaframe/com1DFA/com1DFA.py (hunk 3 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:193]: currently the inputSimfiles ara checked in the end

File: avaframe/com1DFA/com1DFA.py [hunk 6]
```diff
@@ -111,147 +151,148 @@ def com1DFAMain(avalancheDir, cfgMain, cfgFile=''):
...
-                reportDictList.append(reportDict)
 
-                # create hash to check if configuration didn't change
-                simHashFinal = cfgUtils.cfgHash(cfgFinal)
-                if simHashFinal != simHash:
-                    cfgUtils.writeCfgFile(avalancheDir, com1DFA, cfg, fileName='%s_butModified' % simHash)
-                    message = 'Simulation configuration has been changed since start'
-                    log.error(message)
-                    raise AssertionError(message)
+def com1DFAPostprocess(simDF, tCPUDF, simDFOld, cfg, cfgMain, dem, reportDictList):
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py]: renamed

File: avaframe/com1DFA/com1DFA.py (hunk 3 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:190]: leave check in

File: avaframe/com1DFA/com1DFA.py (hunk 2 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py]: rename to exisitng

**Review by fso42 — APPROVED** [review]

**Comment by fso42** [comment]
Results of standardtests are ok

**Pull request merged**